signac/contrib/project.py

```diff
@@ -1901,11 +1902,31 @@ def to_dataframe(self, sp_prefix='sp.', doc_prefix='doc.'):
         """
         import pandas
 
+        class _IncludeExcludeContainer(Container):
```

Comment:
Using a dedicated class -- even if internal -- here seems a little bit like overkill to me. :/

Is a simple check as in `(whitelist is None or key in whitelist) and (blacklist is None or key not in blacklist)` not sufficient?

Reply:
I had that at first but decided this might be preferred because that logic is repeated for sp and doc keys, _and_ the inputs need to be converted to sets (the inputs could be iterables). It ends up being around the same number of lines total but I can change it if you're opposed to this.

Reply:
Done.